src/test/scala/scorex/core/network/peer/PeerManagerSpec.scala

```diff
@@ -26,7 +26,7 @@ class PeerManagerSpec extends FlatSpec with Matchers {
     implicit val defaultSender: ActorRef = p.testActor
 
     val settings = ScorexSettings.read(None)
-    val timeProvider = new NetworkTimeProvider(settings.ntp)
+    val timeProvider = new NetworkTimeProvider(settings.ntp)(system.dispatcher)
```

Comment:
Ohnoes! this was a really bad idea to use Akka's thread pull for the long operations! Sorry for that. Actually we need to introduce the special ExecutionContext for the time consuming operations, like Akka authors advise: https://doc.akka.io/docs/akka/2.5/dispatchers.html#Blocking_Needs_Careful_Management

Reply:
We can even embrace the context lookup for these time-consuming tasks

`implicit val executionContext: ExecutionContext = context.system.dispatchers.lookup("my-blocking-dispatcher")`